Makefile

```diff
@@ -470,6 +471,25 @@ ci-job-tools: ci-setup-tools
 	$(if $(CI_JOB_TOOLS),$(call ci_job_tools))
 
 
+.PHONY: ci-setup-miri
+ci-setup-miri:
+	@rustup component list | grep miri | grep -q installed || rustup component add miri
+
+.PHONY: ci-job-miri
+ci-job-miri: ci-setup-miri
+	@# This is highly experimental at the moment, so just run miri in the
+	@# places it's known to work and there are any tests.
+	@#
+	@# Hangs forever during `Building` for this one :shrug:
+	@#cd libraries/tock-register-interface && CI=true cargo miri test
+	@cd kernel && CI=true cargo miri test
+	@cd arch/cortex-m && CI=true cargo miri test
+	@cd arch/cortex-m0 && CI=true cargo miri test
+	@cd arch/cortex-m3 && CI=true cargo miri test
+	@cd arch/cortex-m4 && CI=true cargo miri test
```

Comment:
why not include arch/rv32i? or just use list_archs.sh the same way you used list_chips.sh?

Reply:
Good point; fixed.